### PR TITLE
fix save state test

### PR DESCRIPTION
**Pull request by MalakBena**

### CODE REVIEW

**Review by ek15kim — APPROVED** [review]

File: src/extraction_lambda/extraction/monitor.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from datetime import datetime
+from datetime import datetime, timezone
```
Comment (Newrotic) [inline src/extraction_lambda/extraction/monitor.py:1]: No need for timezone import

File: test/extraction_lambda/test_monitor.py [hunk 2]
```diff
@@ -87,18 +87,20 @@ def test_get_current_state_returns_0_if_stats_json_missing_key(s3, monitor):
 
 
 def test_save_state_saves_new_state_to_s3_bucket(s3, monitor):
-    with patch('extraction.monitor.datetime') as mock_date:
-        mock_date.now.return_value = datetime.datetime(2023, 3, 27, 1, 2, 3, 4)
+    with patch('extraction.monitor.Monitor.get_utc_timestamp') as mock_timestamp:
+        # mock_date.now.return_value = datetime.datetime(
```
Comment (Newrotic) [inline test/extraction_lambda/test_monitor.py:91]: This could work with a mock like this: datetime.now().replace().timestamp.return_value = ...

**Review by Newrotic — COMMENTED** [review]
Please check comments

File: src/extraction_lambda/extraction/monitor.py (hunk 1 above)
Comment (Newrotic) [inline src/extraction_lambda/extraction/monitor.py:1]:
```suggestion
from datetime import datetime
```

File: test/extraction_lambda/test_monitor.py (hunk 2 above)
Comment (Newrotic) [inline test/extraction_lambda/test_monitor.py:91]:
```suggestion
```

File: test/extraction_lambda/test_monitor.py [hunk 3]
```diff
@@ -87,18 +87,20 @@ def test_get_current_state_returns_0_if_stats_json_missing_key(s3, monitor):
...
+        # 2023, 3, 27, 1, 2, 3, 4).replace(tzinfo=datetime.timezone.utc)
```
Comment (Newrotic) [inline test/extraction_lambda/test_monitor.py:92]:
```suggestion
```

**Review by Newrotic — CHANGES_REQUESTED** [review]

**Review by Newrotic — APPROVED** [review]
LGTM

**Pull request merged**